dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -256,6 +292,322 @@ public ViewAsPageStatus getViewAsStatus(final HttpServletRequest request,
+                                + ", has been skipped. Max contentlet: "    + container.getMaxContentlets()
+                                + ", has been overcome for the container: " + containerId);
+                        continue;
+                    }
+
+                    containerUuidPersona
+                            .computeIfAbsent(
+                                    containerId + uniqueUUIDForRender + personalizedContentlet.getPersonalization(),
+                                    k -> Lists.newArrayList())
+                            .add(personalizedContentlet.getContentletId());
+//                    contextMap.put("EDIT_CONTENT_PERMISSION" + contentlet.getIdentifier(),
+//                            permissionAPI.doesUserHavePermission(contentlet, PERMISSION_WRITE, user));
+
+                    // todo check if needed
+//                    this.widgetPreExecute(contentlet);
+//                    this.addAccrueTags(contentlet);
+
+                    if (personalizedContentlet.getPersonalization().equals(includeContentFor)) {
+
+                        final Map<String, Object> contentPrintableMap = Try
+                                .of(() -> ContentletUtil.getContentPrintableMap(user, contentlet))
+                                .onFailure(f -> Logger.warn(this.getClass(), f.getMessage())).getOrNull();
+                        if (contentPrintableMap == null) {
+
+                            continue;
+                        }
+
+                        contentPrintableMap.put("contentType", contentlet.getContentType().variable());
+                        personalizedContentletMap.add(contentPrintableMap);
+                        contentletIncludedCount++;
+                    }
+                }
+
+                contentMaps.put(CONTAINER_UUID_PREFIX + uniqueUUIDForRender, personalizedContentletMap);
+            }
+
+//            for (final Map.Entry<String, List<String>> entry : containerUuidPersona.entrySet()) {
+//
+//                contextMap.put("contentletList" + entry.getKey(), entry.getValue());
+//                contextMap.put("totalSize"      + entry.getKey(), entry.getValue().size());
+//            }
+
+            raws.add(new ContainerRaw(container, containerStructures, contentMaps));
+        }
+
+        return raws;
+    }
+
+    private boolean isLive(final HttpServletRequest request, final PageMode mode) {
+        return (request == null || request.getSession(false) == null
+                || request.getSession().getAttribute("tm_date") == null) && mode.showLive;
+    }
+
+    private String getPersonaTagToIncludeContent(final HttpServletRequest request, final Set<String> personalizationsForPage) {
+        IPersona iPersona = null;
+
+        if (request != null) {
+            final Optional<Visitor> visitor = APILocator.getVisitorAPI().getVisitor(request);
+            iPersona = visitor.isPresent() && visitor.get().getPersona() != null ? visitor.get().getPersona() : null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6090726)